### PR TITLE
cargo: update msrv to 1.75

### PR DESCRIPTION
**Pull request by prestist**
Additionally run `cargo update` to update all dep's.

### CODE REVIEW

**Comment by prestist** [comment]
Due to nightly not compiling it looks like we have a chicken/egg issue with the lint-ing. I am going to update this pr to include the change from this, and deal with the fallout from the merge of this change later. https://github.com/coreos/repo-templates/pull/230

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Afterburn
 
 [![crates.io](https://img.shields.io/crates/v/afterburn.svg)](https://crates.io/crates/afterburn)
-![minimum rust 1.71](https://img.shields.io/badge/rust-1.71%2B-orange.svg)
+![minimum rust 1.75](https://img.shields.io/badge/rust-1.71%2B-orange.svg)
```
Comment (tormath1) [inline README.md]:
```suggestion
![minimum rust 1.75](https://img.shields.io/badge/rust-1.75%2B-orange.svg)
```

**Review by tormath1 — APPROVED** [review]
LGTM.

File: README.md (hunk 1 above)
Comment (prestist) [reply README.md]: Ah good catch! Thank you

**Review by travier — APPROVED** [review]
LGTM

**Pull request merged**